test/functional-tests/include/ParameterFramework.hpp

```diff
@@ -0,0 +1,135 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#pragma once
+
+#include "Config.hpp"
+#include "ConfigFiles.hpp"
+
+#include <ParameterMgrFullConnector.h>
+
+/** Wrapper around the Parameter Framework to throw exceptions on errors and
+ *  have more user friendly methods.
+ */
+class ParameterFramework : private ConfigFiles, private CParameterMgrFullConnector
+{
+private:
+    using PF = CParameterMgrFullConnector;
+
+    template <class Type>
+    using Getter = Type (PF::*)() const;
+    template <class Type>
+    using Setter = void (PF::*)(Type);
+
+    template <class PType, Getter<PType> getter, Setter<PType> setter>
+    struct Prop {
+        using Type = PType;
+        static Type get(const PF &pf) {
+            return (pf.*getter)();
+        }
+        static void set(PF &pf, Type value) {
+            (pf.*setter)(value);
+        }
+    };
+
+    template <class Type>
+    using MayFailSetter = bool (PF::*)(Type, std::string &);
+
+    template <class PType, Getter<PType> getter, MayFailSetter<PType> setter>
+    struct MayFailProp {
+        using Type = PType;
+        static Type get(const PF &pf) {
+            return (pf.*getter)();
+        }
+        static void set(PF &pf, Type value) {
+            mayFailCall(pf, setter, value);
+        }
+    };
+
+public:
+    using FailureOnMissingSubsystem =
+        Prop<bool, &PF::getFailureOnMissingSubsystem, &PF::setFailureOnMissingSubsystem>;
+    using FailureOnFailedSettingsLoad =
+        Prop<bool, &PF::getFailureOnFailedSettingsLoad, &PF::setFailureOnFailedSettingsLoad>;
+    using ForceNoRemoteInterface =
+        Prop<bool, &PF::getForceNoRemoteInterface, &PF::setForceNoRemoteInterface>;
+    using SchemaFolderLocation =
+        Prop<const std::string &, &PF::getSchemaFolderLocation, &PF::setSchemaFolderLocation>;
+    using Validate =
+        Prop<bool, &PF::getValidateSchemasOnStart, &PF::setValidateSchemasOnStart>;
+    using RawValueSpace =
+        Prop<bool, &PF::isValueSpaceRaw, &PF::setValueSpace>;
+    using HexOutputFormat =
+        Prop<bool, &PF::isOutputRawFormatHex, &PF::setOutputRawFormat>;
+    using TuningMode =
+        MayFailProp<bool,  &PF::isTuningModeOn, &PF::setTuningMode>;
+    using Autosync =
+        MayFailProp<bool, &PF::isAutoSyncOn, &PF::setAutoSync>;
+    using Logger =
+        Prop<ILogger *, nullptr, &PF::setLogger>;
+
+
+    ParameterFramework(const Config &config = Config()) :
+        ConfigFiles(config),
+        PF(getPath()) {}
+
+    void start() {
+        mayFailCall(&PF::start);
+    }
+
+    /** @name Forwarded methods
+     * Forward those methods without modification as they are no failure to
+     * forward as exception and are already ergonomic.
```

Comment:
This sentence does not make much sense to me... First, I suppose there is a typo: "as they are" -> "since there are"; but even so, I have a hard time understanding this sentence.